src/cashweb/wallet/index.ts

```diff
@@ -296,64 +290,68 @@ export class Wallet {
   async updateAllOutpoints() {
     // Check HD Wallet
     const utxos = [...this.storage.getUtxoMap().values()]
-    await P.map(utxos, utxo => this.fixUtxo(utxo), { concurrency: 5 })
+    await this.fixUtxos(utxos)
   }
 
-  async checkOutpoint(utxo: Utxo) {
-    const utxoId = calcUtxoId(utxo)
-    console.log(`Checking UTXO ${utxoId}`)
-    const client = await this.electrumClientPromise
-    assert(client, 'missing client in fixOutpointId')
-    const scriptHash = toElectrumScriptHash(utxo.address)
-    const elOutputs = await client.request(
-      'blockchain.scripthash.listunspent',
-      scriptHash,
-    )
-    assert(
-      elOutputs,
-      `Null response from blockchain.scripthash.listunspent for ${scriptHash}`,
-    )
-    if (elOutputs instanceof Error) {
-      return false
-    }
-    assert(
-      elOutputs instanceof Array,
-      'output of blockchain.scripthash.listunspent was not an array!',
+  async checkAreOutpointsUnspent(utxos: Utxo[]): Promise<boolean[]> {
+    const utxoIds = utxos.map(calcUtxoId)
+    console.log(`Checking UTXOs ${utxoIds}`)
+    const chronikClient = this.chronikClient
+    assert(chronikClient, 'missing client in checkAreOutpointsUnspent')
+    const utxoStates = await chronikClient.validateUtxos(
+      utxos.map(utxo => ({
+        txid: utxo.txId,
+        outIdx: utxo.outputIndex,
+      })),
     )
-    const unspentOutputs = elOutputs as ElectrumListUnspentUtxo[]
-    if (
-      !unspentOutputs.some(output => {
-        return (
-          output.tx_hash === utxo.txId && output.tx_pos === utxo.outputIndex
-        )
-      })
-    ) {
-      console.log(`UTXO missing on-chain, deleting missing UTXO ${utxoId}`)
-      this.storage.deleteById(utxoId)
+    return utxoStates.map((utxoState, idx) => {
+      switch (utxoState.state) {
+        case 'UNSPENT':
+          console.log('UTXO is valid', utxoIds[idx])
+          return true
```

Comment:
This seems to return a simple array. If you want to make it not delete UTXOs here, then it probably should return the UTXO id to allow callers more easily clean them up.